modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -23,9 +23,15 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
     .pickBy((_p, ctx) => input.includedContexts.includes(ctx))
     .entries()
     .map(([name, ctx]) => ({ ...ctx, name }))
-    .maxBy(ctx => ctx.confidence)!
+    .maxBy(ctx => ctx.confidence)! || {
+    name: 'global',
+    confidence: 1.0,
+    oos: 0.0,
+    intents: []
+  }
+
+  const noneIntent = { label: NONE_INTENT, confidence: mostConfidentCtx?.oos || 1.0, slots: {}, extractor: '' }
```

Comment:
can't you rm the `?` and the `|| 1.0`

Reply:
`mostConfidentCtx` is undefined not `oos`